x/oracle/keeper/msg_server_test.go

```diff
@@ -0,0 +1,132 @@
+package keeper_test
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/provenance-io/provenance/x/oracle/keeper"
+	"github.com/provenance-io/provenance/x/oracle/types"
+)
+
+func (s *KeeperTestSuite) TestUpdateOracle() {
+	authority := s.app.OracleKeeper.GetAuthority()
+	event := sdk.NewEvent(
+		sdk.EventTypeMessage,
+		sdk.NewAttribute(sdk.AttributeKeyModule, types.ModuleName),
+	)
+
+	tests := []struct {
+		name  string
+		req   *types.MsgUpdateOracleRequest
+		res   *types.MsgUpdateOracleResponse
+		event *sdk.Event
+		err   string
+	}{
+		{
+			name: "failure - authority does not match module authority",
+			req: &types.MsgUpdateOracleRequest{
+				Address:   "cosmos1w6t0l7z0yerj49ehnqwqaayxqpe3u7e23edgma",
+				Authority: "cosmos1w6t0l7z0yerj49ehnqwqaayxqpe3u7e23edgma",
+			},
+			res: nil,
+			err: fmt.Sprintf("expected authority %s got cosmos1w6t0l7z0yerj49ehnqwqaayxqpe3u7e23edgma: unauthorized", authority),
+		},
+		{
+			name: "success - oracle is updated",
+			req: &types.MsgUpdateOracleRequest{
+				Address:   "cosmos1w6t0l7z0yerj49ehnqwqaayxqpe3u7e23edgma",
+				Authority: authority,
+			},
+			res:   &types.MsgUpdateOracleResponse{},
+			event: &event,
+		},
+	}
+
+	for _, tc := range tests {
+		s.Run(tc.name, func() {
+			res, err := s.msgServer.UpdateOracle(s.ctx, tc.req)
+
+			if tc.event != nil {
+				s.Assert().Equal(1, len(s.ctx.EventManager().Events()), "should emit the correct number of events")
+				s.Assert().Equal(*tc.event, s.ctx.EventManager().Events()[0], "should emit the correct event")
+			} else {
+				s.Assert().Equal(0, len(s.ctx.EventManager().Events()), "should not emit events")
```

Comment:
`.Empty` would be a better assertion here. That way, if it fails, the failure message will include the events which will likely help identify why it failed.